### PR TITLE
HPCC-20878 Enable resourceConditionalActions (stalled)

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by ghalliday** [comment]
Pushed to run regression suite.

**Comment by richardkchapman** [comment]
I think there may be a jira open regarding the need to implement something in Roxie to support this

**Comment by ghalliday** [comment]
Currently some of the examples are much larger - so I need to fix some code generator problem.  However this (and some linked issues) are causing real problems with queries, so probably need to address them sooner rather than later.  (They have been on the radar for years.)

**Comment by ghalliday** [comment]
rebased to rerun smoke test

File: ecl/hqlcpp/hqlttcpp.cpp [hunk 1]
```diff
@@ -8534,7 +8542,8 @@ bool AutoScopeMigrateInfo::doAutoHoist(IHqlExpression * transformed, bool minimi
         return false;
     }
 
-    if (firstUseIsConditional && firstUseIsSequential)
+    if (firstUseIsConditional)
+    //if (firstUseIsConditional && firstUseIsSequential)
```
Comment (richardkchapman) [inline ecl/hqlcpp/hqlttcpp.cpp:8546]: Should this commented line be removed?

File: roxie/ccd/ccdserver.cpp [hunk 2]
```diff
@@ -11969,14 +11970,14 @@ class CRoxieServerDiskWriteActivityFactory : public CRoxieServerMultiOutputFacto
 
     virtual bool isSink() const
     {
-        return numOutputs == 0 && (kind==TAKspillwrite || !isTemp); // MORE - check with Gavin if this is right if not a temp but reread in  same job...
+        return numOutputs == 0 && (kind==TAKspillwrite || (!isTemp && isRoot)); // MORE - check with Gavin if this is right if not a temp but reread in  same job...
```
Comment (richardkchapman) [inline roxie/ccd/ccdserver.cpp:11973]: perhaps the comment can be removed?

**Review by richardkchapman — COMMENTED** [review]
Some minor comments. Would need squashing too.

**Comment by ghalliday** [comment]
Yes.  Also checkboxes not filled in etc.  It isn't ready for review yet - it needs carefully going through the differences in generated queries and investigating the impact.  Some queries grew (one by 100%), others shrank (several by ~50%).  Ideally I would keep the shrinkages, but not keep the growth!

**Comment by ghalliday** [comment]
Stalled for too long, closing.

**Pull request closed without merging**